### PR TITLE
Avoid unbound variables and other surprises in ironic-common

### PR DESCRIPTION
**Pull request by elfosardo**

### CODE REVIEW

File: scripts/ironic-common.sh [hunk 1]
```diff
@@ -1,3 +1,7 @@
+#!/usr/bin/bash
+
+set -euxo pipefail
+
 function get_provisioning_interface() {
   if [ -n "${PROVISIONING_INTERFACE}" ]; then
```
Comment (dtantsur) [inline scripts/ironic-common.sh:6]: I suspect this needs the same treatment

Comment (elfosardo) [reply scripts/ironic-common.sh:6]: yep, and there are a couple more :/

**Comment by elfosardo** [comment]
/retest
"FAIL - node-0 Baremetalhost status OK
       expected OK, got null" lol ?

**Comment by elfosardo** [comment]
centos repo failure Errors during downloading metadata for repository 'centos-nfv-ovs' :/

**Comment by elfosardo** [comment]
/hold
I'll rebase on https://github.com/metal3-io/ironic-image/pull/392

File: scripts/ironic-common.sh [hunk 2]
```diff
@@ -22,7 +26,7 @@ export LISTEN_ALL_INTERFACES="${LISTEN_ALL_INTERFACES:-"true"}"
 # Wait for the interface or IP to be up, sets $IRONIC_IP
 function wait_for_interface_or_ip() {
   # If $PROVISIONING_IP is specified, then we wait for that to become available on an interface, otherwise we look at $PROVISIONING_INTERFACE for an IP
-  if [ ! -z "${PROVISIONING_IP}" ];
+  if [ ! -z "${PROVISIONING_IP:-}" ];
```
Comment (Rozzii) [inline scripts/ironic-common.sh:29]: I think ! -z can be shortened to -n

File: scripts/ironic-common.sh [hunk 3]
```diff
@@ -31,7 +35,7 @@ function wait_for_interface_or_ip() {
       sleep 1
     done
   else
-    until [ ! -z "${IRONIC_IP}" ]; do
+    until [ ! -z "${IRONIC_IP:-}" ]; do
```
Comment (Rozzii) [inline scripts/ironic-common.sh:38]: I think ! -z can be shortened to -n

File: scripts/configure-ironic.sh [hunk 4]
```diff
@@ -35,7 +37,7 @@ wait_for_interface_or_ip
 export IRONIC_BASE_URL="${IRONIC_SCHEME}://${IRONIC_URL_HOST}:${IRONIC_ACCESS_PORT}"
 export IRONIC_INSPECTOR_BASE_URL="${IRONIC_INSPECTOR_SCHEME}://${IRONIC_URL_HOST}:${IRONIC_INSPECTOR_ACCESS_PORT}"
 
-if [ ! -z "${IRONIC_EXTERNAL_IP}" ]; then
+if [ ! -z "${IRONIC_EXTERNAL_IP:-}" ]; then
```
Comment (Rozzii) [inline scripts/configure-ironic.sh:40]: I think ! -z can be shortened to -n

**Review by Rozzii — CHANGES_REQUESTED** [review]

File: scripts/ironic-common.sh (hunk 2 above)
Comment (elfosardo) [reply scripts/ironic-common.sh:29]: yep, I'll change that, thanks for the comment

**Comment by Rozzii** [comment]
It is a NIT, but I would actually prefer the default values to be set at the beginning of the script not in the test commands.

**Review by Rozzii — CHANGES_REQUESTED** [review]
It is a NIT, but I would actually prefer the default values to be set at the beginning of the script not in the test commands. @elfosardo

**Comment by elfosardo** [comment]
@Rozzii I see what you mean, I agree
I'll push a new commit with that

**Pull request merged**